components/ai_chat/renderer/ai_chat_resource_sniffer_throttle_unittest.cc

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
```

Comment:
I'm not entirely sure, but it looks like browser tests would be easier to write, also they can test some near to real api calls.

Reply:
I'm following upstreams example from `MimeSniffingThrottle`. And these tests seem successful in testing the parts we care about - whether the throttle was created and whether the delegate is called as expected. I think the team prefers unit tests to browser tests, in general - less flakey and more performant. Any issue with this?